posts/2021-07-29-Rust-1.54.0.md

```diff
@@ -0,0 +1,74 @@
+---
+layout: post
+title: "Announcing Rust 1.54.0"
+author: The Rust Release Team
+release: true
+---
+
+# Announcing Rust 1.54.0
+
+The Rust team is happy to announce a new version of Rust, 1.54.0. Rust is a programming language empowering everyone
+to build reliable and efficient software.
+
+If you have a previous version of Rust installed via rustup, getting Rust
+1.54.0 is as easy as:
+
+```console
+rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`][install]
+from the appropriate page on our website, and check out the
+[detailed release notes for 1.54.0][notes] on GitHub.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: XXX
+
+## What's in 1.54.0 stable
+
+### Attributes can call macros!()
+
+Rust 1.54 supports invoking function-like macros inside attributes. One notable use case for this is including documentation from other files into Rust doc comments. For example, if your project's README represents a good documentation comment, you can use `include_str!` to directly incorporate the contents. Previously, various workarounds allowed similar functionality, but from 1.54 this is much more ergonomic.
+
+
+```rust=
+#![doc = include_str!("README.md")]
+```
+
+Macros can be nested inside the attribute as well, for example to include content generated by a build script:
+
+```rust
+#[path = concat!(env!("OUT_DIR"), "/generated.rs")]
+mod generated;
+```
+
+Read [here](https://github.com/rust-lang/rust/pull/83366) for more details.
+
+### wasm32 intrinsics stabilized
+
+A number of intrinsics for the wasm32 platform have been stabilized, which gives access to the SIMD instructions in WebAssembly.
+
+Notably, unlike the previously stabilized `x86` and `x86_64` intrinsics, these do not have a safety requirement to only be called when the appropriate target feature is enabled. This is because WebAssembly was written from the start to validate code safely before executing it, so instructions are guaranteed to be decoded correctly (or not at all).
+
+This means that we can expose some of the intrinsics as entirely safe functions, for example [`v128_bitselect`](https://doc.rust-lang.org/beta/core/arch/wasm32/fn.v128_bitselect.html). However, there are still some intrinsics which are unsafe, such as [`v128_load`](https://doc.rust-lang.org/beta/core/arch/wasm32/fn.v128_load.html).
```

Comment:
```suggestion
This means that we can expose some of the intrinsics as entirely safe functions, for example [`v128_bitselect`](https://doc.rust-lang.org/beta/core/arch/wasm32/fn.v128_bitselect.html). However, there are still some intrinsics which are unsafe because they use raw pointers, such as [`v128_load`](https://doc.rust-lang.org/beta/core/arch/wasm32/fn.v128_load.html).
```

Reply:
I intentionally omitted this in case there were other constraints -- the stable docs likely won't be up to date in time for the release with safety comments. There was some discussion on https://rust-lang.zulipchat.com/#narrow/stream/219381-t-libs/topic/wasm32.20intrinsics.20safety.20docs and https://github.com/rust-lang/stdarch/pull/1184 has been merged, but we're not expecting backports to 1.54 right now.

Reply:
It looks like from a quick skim this is indeed accurate in broad strokes (the more precise thing needed is valid memory for N bytes, doesn't need to be aligned), though, so accepting.